### ISSUE TITLE
Prioritze payload | Get two payloads or change order of recieved payloads

### ISSUE THREAD

**Issue by noraroxette** [jenkinsci/generic-webhook-trigger-plugin#252]
Hi!

We have a job in Jenkins which we want to trigger on push, but also on pull request opened and pull request branch updated (from Bitbucket). We use the Generic Webhook Trigger plugin with a token.

However, by doing this, a push to the pull request branch also fires the regular push webhook (makes sense as they are both in fact a push event). Whenever there is an event related to Pull Request, that is the payload we want, but we only receive the "repo:refs_changed" event-payload. It seems that Jenkins only recieves the payload from one of the triggers, the push event and not the pull request branch updated.

```
10:10:00  Generic Cause
10:10:00  Generic Cause
10:10:00  GenericWebhookEnvironmentContributor
10:10:00   Received:
10:10:00  
10:10:00  {"eventKey":"repo:refs_changed",.......}
10:10:00  ---continue with something
```

But as you can see on this log from jenkins, it has been triggered twice, which is correct (the push, and pull request branch updated event). Is there any way to add a delay to the push event on the webhook or somehow change the order the payloads are sent from Bitbucket so that we can continue or Jenkinsjob with the payload we want (the pull request payload if this exist)? Or any way to get both?

Thank you for this plugin :)

**Comment by tomasbjerre** [comment]
See this if you want each individual event to trigger one job:
https://github.com/jenkinsci/generic-webhook-trigger-plugin#trigger-exactly-one-build

If you want to override the delay, there is a *Override Quiet Period* option. But I don't think that solves your problem.

**Comment by tomasbjerre** [comment]
Yes, I can confirm that. But if you do as I try to explain in the readme, you will not have that problem. Just pic something unique from the payload, with jsonpath, and assign it to a variable in the plugin. Make the job parameterized with a parameter having same name as the variable.

**Comment by tomasbjerre** [comment]
I released `1.86.1` now and it should solve the problem. See changelog. Open issue again if not working.
